### PR TITLE
Expose all Taxon parents

### LINKED ISSUE #485: Species detail links are often broken
**Steps to reproduce:**
1. Go to https://app.preview.insectai.org/projects/18/species/5464
2. Click on the genus label
3. Detail view try load load, but then nothing renders (reason is no item with this ID was found)

We should handle this error more gracefully from the FE side, but also look into the reason. Seems to happen for a lot of records, also when it comes to links from occurrence details. I have only noticed it for the lower ranks though if that can be a clue...

### PR DESCRIPTION
**Pull request by mihow**
This adds a new `parents` attribute to the Taxon object in all related API endpoints. This new attribute is an ordered list of all ranks and is intended to be used instead of the recursive `parent` attribute. 

Most UI views have been updated to use the new `parents` list, except for the species list view! I can't crack that one.

One `parent` is still available if needed. 

This enables the bulk ID actions to be much more useful because you can select multiple occurrences and choose the taxon rank that is common among all of them (family, superfamily, lepidoptera, etc)

Also fixes #485, you can click on any rank and the species detail page will not 404.

image

image

image

```json
{
"id": 1211,
"name": "Calliteara pudibunda",
"rank": "SPECIES",
"parent": {
"id": 1210,
"name": "Calliteara",
"rank": "GENUS",
"details": "http://localhost:8000/api/v2/taxa/1210.json",
"user_permissions": [
"update",
"delete"
]
},
"parents": [
{
"id": 1,
"name": "Lepidoptera",
"rank": "ORDER"
},
{
"id": 7743,
"name": "Noctuoidea",
"rank": "SUPERFAMILY"
},
{
"id": 50,
"name": "Erebidae",
"rank": "FAMILY"
},
{
"id": 8554,
"name": "Lymantriinae",
"rank": "SUBFAMILY"
},
{
"id": 11319,
"name": "Orgyiini",
"rank": "TRIBE"
},
...
```

### CODE REVIEW

**Comment by mihow** [comment]
@annavik Can you help update the species list view to get the ranks back? I can't track that one down.

image

File: ui/src/data-services/models/taxa.ts [hunk 1]
```diff
@@ -13,26 +27,26 @@ export class Taxon {
...
-      ranks.push({
-        id: `${current.id}`,
-        name: current.name,
-        rank: current.rank,
-      })
-      current = current.parent
+    if (taxon.parents) {
+      this.ranks = taxon.parents
+    } else if (taxon.parent?.parents) {
+      // TODO: Update this when species list is returning parents similar to other endpoints
```
Comment (annavik) [inline ui/src/data-services/models/taxa.ts:38]: @mihow seems in the species list case, data structure looked slightly different. I could still access all parents though from the `parent` object. We can skip this once we streamline the format :)

File: ui/src/data-services/models/taxa.ts [hunk 2]
```diff
@@ -1,10 +1,24 @@
 export type ServerTaxon = {
-  id: number
+  id: string
   name: string
   rank: string
   parent?: ServerTaxon
+  parents?: ServerTaxon[]
 }
 
+const SORTED_RANKS = [
```
Comment (annavik) [inline ui/src/data-services/models/taxa.ts:9]: @mihow I noticed the ranks were not always sorted (you can for example see this on the occurrence table screenshot you posted in PR description). I added some client side sorting for now, based on this list of ranks. I hope it's correct!

We can skip this if backend plan to sort, but leaving it for now.

**Comment by annavik** [comment]
I went on and updated the rank appearance in table view, to show max 3 ranks (least detailed and most detailed):

Skärmavbild 2024-08-06 kl  10 17 19

Also tweaked the row space, for the case when we need the wrap behaviour:

Skärmavbild 2024-08-06 kl  10 23 55

**Comment by annavik** [comment]
After merging this I came to think of #421 and updated the compact logic to:

Skärmavbild 2024-08-06 kl  10 58 13

See commit https://github.com/RolnickLab/ami-platform/commit/b611a2705e12f537957c1e0951f1d2a7deb6b924

**Pull request merged**